### ISSUE TITLE
FFNNParameters: dependency-free feed-forward networks from layer sizes and activation names

### ISSUE THREAD

**Issue by manuhuth** [manuhuth/NoLimits.jl#261]
## Motivation

Declaring a feed-forward network today requires constructing a backend chain before the `@Model` block:

```julia
using SimpleChains
chain = SimpleChain(static(1), TurboDense(tanh, 8), TurboDense(identity, 1))
# ...
z = NNParameters(chain; function_name = :NN1, calculate_se = false)
```

That is boilerplate for the overwhelmingly common case (a plain MLP), forces a weak-dependency install, and requires backend API knowledge. It is also a structural obstacle for the language bindings and the planned federated packages: a chain is code, but a feed-forward spec is data. `FFNNParameters((1, 8, 1); activation = "tanh")` can be serialized, validated, and whitelisted (DataSHIELD-style disclosure review), while arbitrary chain-building code cannot.

Proposed usage:

```julia
z = FFNNParameters((1, 8, 8, 1); activation = :tanh, output_activation = :identity,
                   function_name = :NN1, calculate_se = false)
# call sites unchanged: NN1([t], z)[1]
```

- First argument: layer sizes as a Tuple or AbstractVector of Integers, length >= 2 (input dim, hidden dims..., output dim).
- `activation`: hidden-layer activation, accepted as Symbol, String (normalized via `_as_symbol`, consistent with #255), or any callable (e.g. `tanh`, a user closure).
- `output_activation`: transformation of the final layer output, same accepted forms, plus vector-valued transforms such as `:softmax`.

## Why a native core implementation (recommended), not sugar over SimpleChains

The NN forward pass already dispatches on the chain object's type: `_param_spec`/`_nn_axes_template`/`_collect_model_fun!` handle `NNParameters` generically (src/model/FixedEffects.jl:565, :602, :615) and the backend-specific forward lives in NoLimitsSimpleChainsExt / NoLimitsLuxExt, selected by the chain's type (src/model/FixedEffects.jl:620-625). Therefore:

1. Add a small core struct, e.g. `FFNN` (fields: layer sizes, hidden activation, output activation), plus a core forward-pass method in the same dispatch slot the extensions use.
2. `FFNNParameters(sizes; activation, output_activation, kwargs...)` is then a thin constructor returning `NNParameters(FFNN(...); kwargs...)`, inheriting ALL existing plumbing: flat parameter layout, `function_name` registration, `calculate_se` masks, priors, and the random-effects-over-weights pattern used in the SAEM tutorials.

This gives FFNNs with ZERO weak dependencies (works on any install, including headless servers), and avoids the SimpleChains failure modes previously hit in this repo (task-local scratch-buffer aliasing, wrong nested-Dual results at depth > 4). SimpleChains/Lux remain fully supported via `NNParameters` for anything beyond plain MLPs.

## Implementation instructions

1. **Activation registry.** A small `const` table (Constants.jl or next to the FFNN struct) mapping symbols to functions: `:tanh, :relu, :sigmoid` (alias `:logistic`), `:softplus, :identity, :gelu, :swish` for hidden/output, plus output-only `:softmax` (numerically stable: subtract the maximum before exponentiating) and `:logit`. Resolution order: Function passes through; Symbol/String looked up after `_as_symbol`, with an error listing valid names on a miss. `:softmax` as a hidden activation should error (it is a vector transform).
2. **`FFNN` struct + forward pass** (new file under src/model/ or alongside the NN code in FixedEffects.jl, matching repo layout). Parameters are the existing flat layout: for layer i, weight matrix (n_out x n_in) then bias (n_out), concatenated. The forward must:
   - be non-mutating or use buffers safe under ForwardDiff Dual eltypes (accumulator eltype promoted from BOTH input and parameter eltypes; use `convert(T, x)` not `T(x)` per repo convention);
   - never reuse task-local scratch across nested calls (the SimpleChains lesson);
   - stay Enzyme-compatible where that does not regress the ForwardDiff hot path (repo policy).
   Total parameter count: sum over layers of n_out*(n_in+1); wire this into `_nn_axes_template`/`_check_nn_flat_layout` the same way the extensions do.
3. **Initialization.** `NNParameters` obtains initial parameters from the chain; give `FFNN` the equivalent: Glorot-uniform weights, zero biases, with an optional `rng`/`init` kwarg mirroring whatever `NNParameters` exposes today. Deterministic given the rng.
4. **`FFNNParameters` constructor.** Validate sizes (Integers, length >= 2, all positive), resolve both activations via the registry, construct `FFNN`, forward remaining kwargs to `NNParameters` unchanged. Export it next to `NNParameters`.
5. **Error messages.** The existing "no forward pass is defined for a chain of type ..." error (src/model/FixedEffects.jl:625) must never fire for `FFNN`; a helpful error should list registry names when an unknown activation is given.

## Testing (existing files where possible)

- Forward-pass correctness: compare `FFNN` output against a hand-computed two-layer example (fixed weights) for each registry activation; softmax output sums to 1.
- AD: ForwardDiff gradient of the forward pass w.r.t. parameters matches FiniteDifferences on a small example; one nested-Dual case (Dual input AND Dual parameters) to pin the SimpleChains regression class.
- Integration: one small mixed-effects fit reusing the existing NN testset pattern with `FFNNParameters` in place of the SimpleChains chain (keep it minimal; every new @Model is compile-budget, so ONE new model at most, placed in the existing NN test file).
- Equivalence: same architecture in SimpleChains and FFNN with identical copied weights produces identical predictions (guards the layout convention), in the existing SimpleChains test file (skip if extension not loaded).
- String/Symbol/Function forms of `activation` produce identical results.

## Documentation

- Docstring with the full registry list.
- The NN tutorial gains a short "no dependencies needed" FFNN variant before the SimpleChains section; mention that SimpleChains/Lux remain the route for custom architectures.

## Non-goals / deferred

- Per-layer activation lists (possible later extension of the same registry; do not build now).
- Convolutional/recurrent layers, dropout, normalization layers - out of scope; `NNParameters` covers advanced cases.
- No change to `NNParameters` itself.
